db/config.cc

```diff
@@ -1225,6 +1225,7 @@ db::config::config(std::shared_ptr<db::extensions> exts)
         "The maximum fraction of cache memory permitted for use by index cache. Clamped to the [0.0; 1.0] range. Must be small enough to not deprive the row cache of memory, but should be big enough to fit a large fraction of the index. The default value 0.2 means that at least 80\% of cache memory is reserved for the row cache, while at most 20\% is usable by the index cache.")
```

Comment:
"`join_cluster` becomes even messier."

Well, I have a plan to clean it up by separating raft topology and non raft topology paths as much as possible, but I need time for this.

Reply:
> Well, I have a plan to clean it up by separating raft topology and non raft topology paths as much as possible, but I need time for this.

Keep in mind that we plan to have the gossip-based topology removed in 2025.2. Maybe there is no need to clean this code up just before.

Reply:
> Keep in mind that we plan to have the gossip-based topology removed in 2025.2

Don't we want to support upgrade from 2023.x? In which version raft topology became mandatory?

Reply:
> In which version raft topology became mandatory?

It didn't yet, but I think the plan is to make it mandatory for 2025.2.

IIUC for 2025.1 you can still upgrade from 2023.1 and then upgrade to raft-topology (which means 2025.1 still supports gossip mode). For 2025.2 we could require to upgrade to raft-topology before upgrading version to 2025.2 by failing on boot if we detect that the cluster is still trying to use gossip mode.